docs/resources/cloud_gateway_transit_gateway.md

```diff
@@ -35,6 +35,28 @@ resource "konnect_cloud_gateway_transit_gateway" "my_cloudgatewaytransitgateway"
       transit_gateway_id = "...my_transit_gateway_id..."
     }
   }
+  awsvpc_peering_gateway = {
```

Comment:
@mheap should there be a `_` between aws and vpc here - `aws_vpc_peering_gateway`? 🤔

Reply:
Yes, I think so

Reply:
Let me check why this might be happening and fix.

Reply:
I think it's because it's all capitals `AWS VPC Peering Gateway`, If it were `AWS Vpc Peering Gateway` it would translate to `aws_vpc_peering_gateway` I think

Reply:
The solution may be to add a `discriminator` with `kind` being the field to choose. Then set the mapping key to `AwsVpcPeeringGateway`

https://github.com/Kong/terraform-provider-konnect/pull/141/files#diff-d910ba2ef878f7db0223a966b81c8b3f3b65027bb39e4431bb05140171eece39R13336-R13338

Reply:
It is due to the capitalization (changing VPC to Vpc seems to fix it), will try the discriminator

Reply:
`kind` is nested under another property though - transit_gateway_attachment_config
https://github.com/Kong/terraform-provider-konnect/blob/b2ce12324b74af61551264b942ccb014bc52c77e/openapi.yaml#L13113
Trying if it works

Reply:
Ah yes, I'm not sure how a discriminator would work here. Maybe we use an overlay to fix the title for generation

Reply:
I'll give `x-speakeasy-name-override` a shot, if not, i'll try an overlay

Reply:
name override seems to have other side effects where required properties are marked optional. Overlay worked, i'll update this PR and raise another on platform api for overlay